workspace/resource_notebook.go

```diff
@@ -272,6 +279,10 @@ func ResourceNotebook() common.Resource {
 				Path:      path,
 				Overwrite: true,
 			}
+			if createNotebook.Format == "" && createNotebook.Language != "" {
+				createNotebook.Format = "SOURCE"
+				d.Set("format", createNotebook.Format)
+			}
```

Comment:
To make this easier to follow, it seems like we should do something like the following:
```go
if createNotebook.Language == "" {
	// Users must not specify `content_base64` without a language, so
	// `source` must be set.
	ext := strings.ToLower(filepath.Ext(d.Get("source").(string)))
	createNotebook.Language = extMap[ext].Language
	createNotebook.Format = extMap[ext].Format
	// Overwrite cannot be used for Dbc format
	createNotebook.Overwrite = extMap[ext].Overwrite
	d.Set("language", createNotebook.Language)
} else {
	createNotebook.Format = "SOURCE"
}
d.Set("format", createNotebook.Format)
```
Is this logic correct?